### DISCUSSION TITLE
SCT in integrated object

### DISCUSSION THREAD

**Discussion by Sa753** [satijalab/seurat#5047] [Q&A] [answered]
Dear team,

Would you please advise on the following, 

If I have an 6 10X samples that I run SCTransform on each of them separately then integrated them into one seurat object. Now I have 3 assays, RNA, SCT and integrated. 

My questions are:
1- If I want to run differential gene expression, can I use the SCT assay instead of normalising the RNA assay and using it?. is there any advantage in using one over the other?

2- If I run gene enrichment , I run it on the highly variable genes instead of the whole matrix, which one should I use the scale.data in the SCT assay or the integrated assay?

Thanks for your help

**Comment by saketkc** [comment, marked as answer]
> 1- If I want to run differential gene expression, can I use the SCT assay instead of normalising the RNA assay and using it?. is there any advantage in using one over the other?

The short answer is, it depends (yes, if the sequencing depths are comparable across the 6 samples). For the longer explanation see [this comment](https://github.com/satijalab/seurat/issues/5009#issuecomment-907426252)

> If I run gene enrichment , I run it on the highly variable genes instead of the whole matrix, which one should I use the scale.data in the SCT assay or the integrated assay?

If you are looking for variable genes, you would use the `scale.data` slot of SCT assay. The variable genes might differ across samples though.

**Reply by Sa753** [reply]
Thank you very much

**Comment by rahulnutron** [comment]
Hi All,

Thanks for opening this discussion!
Here I want to discuss some of my observations while analysing different scRNA-seq datasets. I am going to mention my thoughts and maybe some issues. Please correct me and highlight your suggestions.

I have integrated data processed with scTransform. I also faced some confusion over using the correct assay use for differential gene expression (DEGs) and UMAP visualization across samples (across conditions, split figures) and only across clusters (no condition wise comparison).

For analysing the integrated data in only cluster wise (no across sample or split figure comparison): I find cluster specific differentially expressed genes (DEGs) by setting RNA as default assay and Normalizing it.  Now which one among the a) sequencing depth corrected expression (integrated), b) SCT or c) normalized RNA is used for expression visualization? From top to bottom of the fig below shows the same gene's expression in SCT, integrated and normalized RNA assay.

![image](https://user-images.githubusercontent.com/18325626/132224382-617d3311-abd4-4580-9847-bc890f75e290.png)




In both SCT and RNA the gene shows expression in a left side cluster too other than the top centre one, as well as the scale bar for integrated is quite different. 

**Split figure**
Next, one of the main use of integrated data is to compare DEGs across whole samples (conditions) and cells of different conditions within the same cluster.  I find the DEGs across conditions using normalized RNA, SCT and integrated assays and checked the expression using all the 3. Many DEGs show opposite up/down regulation while using RNA or SCT or integrated. A gene SN identified suing SCT to be down in S4, looks up in S4 condition with integrated and RNA however, correctly it looks down in S4 with SCT. If the expression is experimentally known, its traceable, however for unknown genes it is not simple to decide which one is correct.

I also want to highlight an issue: When I changed the colour scale to yellow-darkred with all other same arguments, some cells doesn't show expression as it was in blue scale (see arrow in RNA).


![image](https://user-images.githubusercontent.com/18325626/132218931-504cf7b9-807e-4d78-b237-8dd9eda00753.png)

Although it is not recommended, I also used integrated assay to find the DEGs across conditions, and it worked as expected (shows similar pattern in the experiment) which show correct upregulation in one condition. Although, we are still testing this.

In conclusion I think to get the DEGs (across sample/condition) normalized RNA (similar sequencing depth) or even integrated (non-comparable sequencing depth) assay can be used. To check the gene expression pattern in split figure, integrated assay can be used. 

For only cluster wise DEGs, normalized RNA assay can be used and to check the gene expression SCT (similar sequencing depth) or integrated (non-comparable sequencing depth) can be used.


I will be grateful if you could correct me if I am wrong and also suggest your thoughts.

Regards,
RS

**Reply by saketkc** [reply]
We do not recommend using integrated assay for DE - these are imputed values and hence do not have direct biological interpretation. The idea of integration is to define cell-to-cell correspondence between two datasets that might have say batch effects originally.

> find cluster specific differentially expressed genes (DEGs) by setting RNA as default assay and Normalizing it. Now which one among the a) sequencing depth corrected expression (integrated), b) SCT or c) normalized RNA is used for expression visualization?

You can use the SCT data slot for performing DE as long as the sequencing depth across the two datasets are comparable. The reasoning for this is in my previous comment. 
You can also use the RNA data slot for performing DE after switching to the RNA slot and running LogNormalization on it.

From the figures, it is not clear if this is a marker gene - I just see one large cluster in the first few figures. The differences between RNA and SCT are expected - some cells can have a large impact due to sequencing depth which is regressed out by SCT. It would help if you also paste the code.

**Reply by rahulnutron** [reply]
Thank you sir for your answer! I will get back with the code. 
The first figure shows whole dataset and the gene express in one of the cluster.

![image](https://user-images.githubusercontent.com/18325626/132236587-8934c227-3652-4223-9390-ed931374cde8.png)

**Reply by rahulnutron** [reply]
Hello, 

also which assay/slot you recommend for split figure visualization after getting the differentially expressed genes? Is it possible to use the same RNA and log normalize values for across condition visualization even if the integration was done by SCT method?
